server/src/utils/validators.ts

```diff
@@ -0,0 +1,32 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import SurveyDBHandler from '../databaseHandlers/surveyDBHandler';
+import { SurveyResultRequest } from '../models/surveyModel';
+
+export const surveyResultRequestValidator = async (
+  requestData: SurveyResultRequest,
+  surveyDBHandler: SurveyDBHandler
+): Promise<string[]> => {
+  const { callId, acsUserId, response } = requestData;
+  const errors: string[] = [];
+  const responseAllowedtypes = ['boolean', 'number', 'string'];
+
+  // Check fields are undefined
+  if (!callId) errors.push('callId must be present');
+  if (!acsUserId) errors.push('acsUserId must be present');
+  if (response === undefined) errors.push('response must be present');
+
+  // Check if each field types are correct
+  if (callId && typeof callId !== 'string') errors.push('callId type must be string');
```

Comment:
is checking if callId is defined necessary here? we are already doing that before this step right

Reply:
I think so. This is because if we don't check if `callId` existence, the `callId type must be string` error will include also and I think it is strange since the `callId` is not even present.